tests/test_util.py

```diff
@@ -41,3 +62,177 @@ class A:
     assert str(record[1].message) == "unsupported attribute 'baz' with value '3' for class 'A'"
     assert d_filtered is not d_orig
     assert d_filtered == {'foo': 1}
+
+def test_sshmanager_get():
+    assert sshmanager != None
+
+def test_sshconnection_get():
+    from labgrid.util.ssh import SSHConnection
+    SSHConnection("localhost")
+
+def test_sshconnection_inactive_raise():
+    from labgrid.util.ssh import SSHConnection
+    con = SSHConnection("localhost")
+    with pytest.raises(ExecutionError):
+        con.run_command("echo Hallo")
+
+def test_sshconnection_connect(connection_localhost):
+    assert connection_localhost.isconnected()
+    assert os.path.exists(connection_localhost._socket)
+
+def test_sshconnection_run(connection_localhost):
+    assert connection_localhost.run_command("echo Hello") == 0
+
+def test_sshconnection_port_forward_add_remove(connection_localhost):
+    port = 1337
+    test_string = "Hello World"
+
+    local_port = connection_localhost.add_port_forward('localhost', port)
+    nc_listen = subprocess.Popen(["nc", "-l", "-p", "1337"], stdout=subprocess.PIPE, stdin=None)
```

Comment:
Why do you use nc instead of opening the socket in python?

Reply:
Because I only thought of nc. I'll rework this to use sockets.